specs/agents/span-links.md

```diff
@@ -0,0 +1,36 @@
+## Span Links
+
+A Span or Transaction MAY link to zero or more other Spans/Transactions that are causally related.
+
+Example use-cases for Span Links:
+
+1. When a single transaction represents the batch processing of several messages, the agent SHOULD be able to link back to the traces that have produced the messages.
+2. When the agent receives a `traceparent` header from outside a trust boundary, it SHOULD restart the trace (creating a different trace id with its own sampling decision) and link to the originating trace.
+3. Close gap for the OTLP intake - [OTel's specification of span links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans)
+
+Spans and Transactions MUST collect links in the `links` array with the following fields on each item:
+- `trace.id`: the id of the linked trace.
+- `span.id`: the id of the linked span or transaction.
+
+Example:
+
+```
+"links": [
+  {"trace": {"id": "traceId1"}, "span": {"id": "spanId1"}},
+  {"trace": {"id": "traceId2"}, "span": {"id": "spanId2"}},
+]
+```
```

Comment:
Per https://github.com/elastic/apm-server/pull/7553/files#diff-7c3116c7517c49da2ce5db0a2d59625298a0c96bb28d3b9127e3a81aed59f1f1R566 it looks like these will be `trace_id` and `span_id`.

Reply:
Hahh, thanks, fixed.